multiphase_sorting/multiphase_sorting/multiphase_sorting.cpp

```diff
@@ -1 +1,302 @@
+#include <algorithm>
+#include <cstdio>
+
 #include "multiphase_sorting.hpp"
+
+void MultiphaseSort::setFileCount(const int value) {
+    if(value <= 2) {
+        std::cerr << "MultiphaseSort::setFileCount: file count must be > 2. File count = 3. . . " << std::endl;
+        m_fileCount = 3;
+    }
+    m_fileCount = value;
+    
+    m_supportFiles.resize(m_fileCount);
```

Comment:
Гарантируется, что при этом ни один из указателей контейнера **не будет** смотреть на выделенную память?

Reply:
Вы имеете в виду то, чтобы новые элементы стали сразу nullptr? Я пробовал делать resize(count, nullptr), но компилятор не видит такой метод

Reply:
Поясняю: утечек памяти не будет?

Reply:
Вроде как он должен инициализировать значением по умолчанию, но даже если он(std::resize) выделит память сам, то сам и должен разобраться с её освобождением, мы вроде сами не трогаем память. Но мне кажется, там будет лежать мусор

Reply:
Сам resize будет выделять память только под указатели.
Проблема в том, что при расширении класса (например, добавлении новых алгоритмов) можно забыть про то, что в начале сортировки нужна аллокация, а в конце - деаллокация. Так что лучше воспользоваться второй перегрузкой std::vector::resize:
```suggestion
    m_supportFiles.resize(m_fileCount, nullptr);
```
В этом случае новые указатели будут гарантированно занулены, что облегчит отлов ошибки в случае неверного их использования. В m_deAllocateFileType тоже лучше занулить указатели после удаления.

Reply:
Да, я так и хотел, но у меня компилятор не видит(фото ниже). У меня была идея занулять указатели после удаления, но потом показалось это лишней операцией, исправлено
<img width="1171" alt="Снимок экрана 2024-03-06 в 18 41 51" src="https://github.com/KVRTeJ/Chuder/assets/144002117/9964d6df-15e4-47ad-a9ed-0aff647dbcd5">

Reply:
Как я понял, ему нужна ссылка на объект?
<img width="361" alt="Снимок экрана 2024-03-06 в 18 45 12" src="https://github.com/KVRTeJ/Chuder/assets/144002117/ec91d3e0-285b-4284-906d-3f9b22de8456">

Reply:
Ссылка на константу. Т.к. у нас вектор указателей, nullptr передать допустимо.